src/lib.rs

```diff
@@ -575,6 +577,22 @@ impl GooseAttack {
     /// ```
     pub fn register_scenario(mut self, mut scenario: Scenario) -> Self {
         scenario.scenarios_index = self.scenarios.len();
+        // Machine names must be unique. If this machine name has already been seen, add an
+        // integer at the end to differentiate.
+        let mut conflicts: u32 = 0;
+        let mut machine_name = scenario.machine_name.to_string();
+        // Inserting into the scenario_machine_names hashset will fail if this is name was
+        // already seen.
+        while !self.scenario_machine_names.insert(machine_name) {
+            // For each conflict increase the counter and try again.
+            conflicts += 1;
+            machine_name = format!("{}_{}", scenario.machine_name, conflicts);
+        }
+        // If there was a conflict, also update the scenario itself.
+        if conflicts > 0 {
+            scenario.machine_name = format!("{}_{}", scenario.machine_name, conflicts);
+        }
```

Comment:
How does the ordering work here? Is it possible that the order of scenarios would change with some minor code change? This would then shuffle them and effectively change machine names. Which is not the end of the world, but could cause confusion.